src/algorithms.jl

```diff
@@ -869,3 +869,16 @@ struct BAOAB{T} <: StochasticDiffEqAlgorithm
   scale_noise::Bool
 end
 BAOAB(;gamma=1.0, scale_noise=true) = BAOAB(gamma, scale_noise)
+
+struct ABOBA{T} <: StochasticDiffEqAlgorithm
```

Comment:
maybe add a docstring that contains the SDE form:
<img width="826" alt="Screen Shot 2023-09-19 at 11 44 02 AM" src="https://github.com/SciML/StochasticDiffEq.jl/assets/42201748/dd5e029c-2ff3-4618-9f86-4fcce0cc8166">

(or to the SDE solver docs https://docs.sciml.ai/DiffEqDocs/stable/solvers/sde_solve/ given that it doesn't match our common diagonal/non-diagonal noise classification exactly?)

Reply:
Or maybe it needs a different problem type?

Reply:
There is already the DynamicalSDEProblem (was introduced by the previous implementation of BAOAB) that is supposed to be solved by those integrators

Reply:
Should I add some alg_compatible() functions to allow only DynamicalSDEProblem to be solved by Dynamical integrators?

Reply:
yes that would be helpful in order to gate it to the right form.

Reply:
I just realize that the DynamicalSDEProblem constructor actually return a SDEProblem so alg_comptable functions won't be useful here

Reply:
But it can dispatch on `prob.problem_type`

Reply:
yeah I think we should use `DynamicalSDEProblem` and it's a good idea to use `alg_compatible` to make sure that only BAOAB, ABOBA, etc. can be used to solve it.